lib/shopify_api/webhooks/handler.rb

```diff
@@ -4,12 +4,29 @@
 module ShopifyAPI
   module Webhooks
     module Handler
+      include Kernel
       extend T::Sig
       extend T::Helpers
-      interface!
+      abstract!
 
-      sig { abstract.params(topic: String, shop: String, body: T::Hash[String, T.untyped]).void }
+      sig do
+        params(data: T::Hash[Symbol, T.untyped]).void
+      end
+      def handle_webhook(data)
+        if use_handle_webhook?
+          raise NotImplementedError, "You must implement the `handle_webhook` method in your webhook handler class."
+        end
+      end
+
+      sig do
+        abstract.params(topic: String, shop: String, body: T::Hash[String, T.untyped]).void
+      end
       def handle(topic:, shop:, body:); end
+
+      sig { returns(T::Boolean) }
+      def use_handle_webhook?
```

Comment:
I wonder if this name is clear enough - is there something we can name it that indicates what it does (i.e. that it takes in the metadata object, or the full metadata)?

We could even take it a bit further: if we need a new method, we could simply test for the presence of that method in the class when we're calling, so `use_handle_webhook?` wouldn't be necessary.

Reply:
Hmm right - I guess we can't have an interface with an optional method, which complicates things indeed. Let's go ahead with this approach then.

Reply:
As discussed, I refactored the handler to have multiple handler modules. 
Developers can implement the new handler if they want they additional data.